chapters/memory.txt

```diff
@@ -3294,6 +3458,86 @@ include::{generated}/api/protos/vkUnmapMemory.txt[]
 include::{generated}/validity/protos/vkUnmapMemory.txt[]
 --
 
+ifdef::VK_EXT_map_memory2[]
+[open,refpage='vkUnmapMemory2EXT',desc='Unmap a previously mapped memory object',type='protos']
+--
+Alternatively, to unmap a memory object once host access to it is no longer
+needed by the application, call:
+
+include::{generated}/api/protos/vkUnmapMemory2EXT.txt[]
+
+  * pname:device is the logical device that owns the memory.
+  * pname:pMemoryUnmapInfo is a pointer to a slink:VkMemoryUnmapInfoEXT
+    structure describing parameters of the unmap.
+
+This function behaves identically to flink:vkUnmapMemory except that it gets
+its parameters via an extensible structure pointer rather than directly as
+function arguments.
+
+include::{generated}/validity/protos/vkUnmapMemory2EXT.txt[]
+--
+
+[open,refpage='VkMemoryUnmapInfoEXT',desc='Structure containing parameters of a memory unmap operation',type='structs']
+--
+The sname:VkMemoryUnmapInfoEXT structure is defined as:
+
+include::{generated}/api/structs/VkMemoryUnmapInfoEXT.txt[]
+
+  * pname:sType is the type of this structure.
+  * pname:pNext is `NULL` or a pointer to a structure extending this
+    structure.
+  * pname:memory is the slink:VkDeviceMemory object to be unmapped.
+
+.Valid Usage
+****
+  * pname:memory must: be currently host mapped
+ifdef::VK_EXT_map_memory_placed[]
+  * If ename:VK_MEMORY_UNMAP_RESERVE_BIT_EXT is set in pname:flags,
+    the <<features-memoryUnmapReserve, pname:memoryUnmapReserve>>
+    must: be enabled
+ifdef::VK_EXT_external_memory_host[]
+  * If ename:VK_MEMORY_UNMAP_RESERVE_BIT_EXT is set in pname:flags, the
+    memory object must: not have been imported from a handle type of
+    ename:VK_EXTERNAL_MEMORY_HANDLE_TYPE_HOST_ALLOCATION_BIT_EXT or
+    ename:VK_EXTERNAL_MEMORY_HANDLE_TYPE_HOST_MAPPED_FOREIGN_MEMORY_BIT_EXT
+endif::VK_EXT_external_memory_host[]
+endif::VK_EXT_map_memory_placed[]
+****
+
+include::{generated}/validity/structs/VkMemoryUnmapInfoEXT.txt[]
+--
+
+ifdef::VK_EXT_map_memory_placed[]
+[open,refpage='VkMemoryUnmapFlagBitsEXT',desc='Bitmask specifying additional parameters of a memory unmap',type='enums']
+--
+Bits which can: be set in slink:VkMemoryUnmapInfoEXT::pname:flags, specifying
+additional properties of a memory unmap, are:
+
+include::{generated}/api/enums/VkMemoryUnmapFlagBitsEXT.txt[]
+
+  * ename:VK_MEMORY_UNMAP_RESERVE_BIT_EXT requests that virtual address
+    range currently occupied by the memory map be remain reserved after the
+    flink:vkUnmapMemory2EXT call completes.
+    Future system memory map operations or calls to flink:vkMapMemory or
+    flink:vkMapMemory2EXT will not return addresses in that range unless
+    the range has since been unreserved by the client or the memory map
+    operation is done with ename:VK_MEMORY_MAP_PLACED_BIT_EXT or the system
```

Comment:
```suggestion
    operation is done with ename:VK_MEMORY_MAP_PLACED_BIT_EXT, or the system
```

Reply:
Same.  The last or is sort of a parenthetical.  It's supposed to read as unreserved || (VK_MAP_PLACED || SYS_MAP_FIXED)

Reply:
I agree with @tobias - the comma disambiguates the grouping of the multiple 'or' clauses. It's somewhat like an 'eats shoots and leaves' exercise.

For complete disambiguation, though, a bullet list is probably better in this case.

Reply:
Ok, I think I've improved it.  It now says:

> Future system memory map operations or calls to flink:vkMapMemory or
> `vkMapMemory2EXT` will not return addresses in that range unless
> the range has since been unreserved by the client or the mapping is
> explicitly placed in that range by calling `vkMapMemory2EXT` with
> `VK_MEMORY_MAP_PLACED_BIT_EXT` or doing the system memory map
> equivalent.

Reply:
If we go with the explicit two bits I mentioned above, we could replace all this "reserve" nonsense and say that it replaces the Vulkan memory map with an inaccessible (`PROT_NONE`) anonymous map.

Reply:
That's better, but I think the (or doing the system memory map equivalent) is something that should be in brackets. The lack of punctuation melts my brain 🙃